aiida_quantumespresso/parsers/parse_raw/cp.py

```diff
@@ -81,54 +88,57 @@ def parse_cp_text_output(data, xml_data):
-                parsed_data['econs'] = [float(this_line.split()[6])]
-            except ValueError:
-                pass
-            try:
-                parsed_data['econt'] = [float(this_line.split()[7])]
-            except ValueError:
-                pass
-            try:
-                parsed_data['vnhh'] = [float(this_line.split()[8])]
-            except (ValueError, IndexError):
-                pass
-            try:
-                parsed_data['xnhh0'] = [float(this_line.split()[9])]
-            except (ValueError, IndexError):
-                pass
-            try:
-                parsed_data['vnhp'] = [float(this_line.split()[10])]
-            except (ValueError, IndexError):
-                pass
-            try:
-                parsed_data['xnhp0'] = [float(this_line.split()[11])]
-            except (ValueError, IndexError):
-                pass
+    try:
+        for count,line in enumerate(reversed(data)):
+            if 'nfi' in line and 'ekinc' in line and 'econs' in line:
+                this_line = data[len(data)-count]
+                try:
+                    parsed_data['ekinc'] = [float(this_line.split()[1])]
+                except ValueError:
+                    pass
+                try:
+                    parsed_data['temph'] = [float(this_line.split()[2])]
+                except ValueError:
+                    pass
+                try:
+                    parsed_data['tempp'] = [float(this_line.split()[3])]
+                except ValueError:
+                    pass
+                try:
+                    parsed_data['etot'] = [float(this_line.split()[4])]
+                except ValueError:
+                    pass
+                try:
+                    parsed_data['enthal'] = [float(this_line.split()[5])]
+                except ValueError:
+                    pass
+                try:
+                    parsed_data['econs'] = [float(this_line.split()[6])]
+                except ValueError:
+                    pass
+                try:
+                    parsed_data['econt'] = [float(this_line.split()[7])]
+                except ValueError:
+                    pass
+                try:
+                    parsed_data['vnhh'] = [float(this_line.split()[8])]
+                except (ValueError, IndexError):
+                    pass
+                try:
+                    parsed_data['xnhh0'] = [float(this_line.split()[9])]
+                except (ValueError, IndexError):
+                    pass
+                try:
+                    parsed_data['vnhp'] = [float(this_line.split()[10])]
+                except (ValueError, IndexError):
+                    pass
+                try:
+                    parsed_data['xnhp0'] = [float(this_line.split()[11])]
+                except (ValueError, IndexError):
+                    pass
+    except IndexError:
+        #when the cp does a cg, the output is different and the parser above does not work
+        pass
```

Comment:
If you know it's because if `cg` maybe you can check this explicitly? Or maybe at least issue a log message that will end up in the report, or add a warning, just to know why we will not find some of the outputs?

Reply:
BTW - is this information also elsewhere? Can we avoid to parse the output file now?